clients/drcachesim/tracer/tracer.cpp

```diff
@@ -2336,7 +2336,14 @@ drmemtrace_client_main(client_id_t id, int argc, const char *argv[])
             DR_ASSERT(false);
 #ifdef UNIX
         /* we want an isolated fd so we don't use ipc_pipe.open_for_write() */
-        int fd = dr_open_file(ipc_pipe.get_pipe_path().c_str(), DR_FILE_WRITE_ONLY);
+        const char *pipe_path = ipc_pipe.get_pipe_path().c_str();
+        if (!dr_file_exists(pipe_path))
+            NOTIFY( 0,
+                    "drmemtrace: creating pipe file at \"%s\". Use \"-offline\" mode if "
```

Comment:
Should it say "attempting to open write end of pipe for online analysis but pipe does not exist"?

Reply:
Maybe with a `WARNING: ` prefix too?